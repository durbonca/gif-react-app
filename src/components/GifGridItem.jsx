import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title}/>
      <div className='title-box'>
        <p>{title}</p>
      </div>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}