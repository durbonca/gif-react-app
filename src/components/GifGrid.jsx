import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && <i className='animate__animated flash'>Cargando...</i>}

            <div className='card-grid'>
                {images.map( ( img, i ) =>
                    (
                        <GifGridItem key={i} {...img} />
                        )
                        )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}