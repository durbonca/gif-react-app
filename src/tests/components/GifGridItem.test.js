import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('GifGridItem component test', () => {
    const title = 'Title';
    const url = 'https://media1.giphy.com/media/l2Jhjvw7QXZW5Qq2U/200w.gif';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('should render a component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un pÃ¡rrafo en el title', () => {
        expect(wrapper.find('p').text().trim()).toBe(title);
    })

    test('debe tener una imagen', () => {
        expect(wrapper.find('img').prop('src')).toBe(url);
        expect(wrapper.find('img').prop('alt')).toBe(title);
    })

    test('debe tener una clase "animate__fadeIn"', () => {
        expect(wrapper.find('div').at(0).hasClass('animate__fadeIn')).toBe(true);
    })
})