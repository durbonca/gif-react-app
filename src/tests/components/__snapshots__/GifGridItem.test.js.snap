// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem component test should render a component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://media1.giphy.com/media/l2Jhjvw7QXZW5Qq2U/200w.gif"
  />
  <div
    className="title-box"
  >
    <p>
      Title
    </p>
  </div>
</div>
`;
